File: 6-Patron/python/FabricaBotasNormales.py
import FabricaBotas
import Bota


class Fabrica_BotasNormales(FabricaBotas):
    def __init__(self, botasNormales):
        super().__init__()
        self.__botasNormales = botasNormales

    def solicitarBotas(self):
        return self.__botasNormales

    def crearBotas(self):
        self.__botasNormales = Bota(100)

File: 6-Patron/php/Fabrica_BotasNormales.php
<?php
include './FabricaBotas.php';
include './Bota.php';
class Fabrica_BotasNormales extends FabricaBotas{
	private $botasNormales;

	public function __construct($botasNormales){
    	$this->botasNormales = $botasNormales;
	}
 	
 	public function solicitarBotas(){
    	return $this->botasNormales;
	}

	public function crearBotas(){
    	$nuevaBotaN = new Bota(100);
    	$this->botasNormales = $nuevaBotaN;
	}

}
?>
